src/number_format.js

```diff
@@ -407,7 +410,8 @@ class NumberFormat extends React.Component {
     const {state, props} = this;
     const {isAllowed} = props;
     const lastValue = state.value;
-    let {formattedValue, value} = this.formatInput(inputValue);
+    const {value} = this.formatInput(inputValue);
```

Comment:
Calling formatInput twice is not a good idea just to maintain const and let.
You can probably store it in some different variable. 
```js
const values = this.formatInput(inputValue);
const {value} = values;
let {formattedValue} = values;
```
Or I will prefer just to disable eslint for this line
  ```js 
 let {formattedValue, value} = this.formatInput(inputValue); // eslint-disable-line prefer-const
```